### ISSUE TITLE
Porgram crash on the AVD of Android Studio when the model has multiple input nodes

### ISSUE THREAD

**Issue by xyannly** [microsoft/onnxruntime#8095]
**Describe the bug**
The model was exported as onnx format and converted to the ort format by following the [doc](https://www.onnxruntime.ai/docs/how-to/mobile/model-conversion.html). The input nodes are defined as "X1":[1, 466] and "X2":[1,52], the output node is "Y": [1,462].
Run the Android project in Android Studio, got the program crash and error log:
<pre>
I/studio.deploy: Finished instrumenting
D/NetworkSecurityConfig: No Network Security Config specified, using platform default
D/NetworkSecurityConfig: No Network Security Config specified, using platform default
D/libEGL: loaded /vendor/lib/egl/libEGL_emulation.so
D/libEGL: loaded /vendor/lib/egl/libGLESv1_CM_emulation.so
D/libEGL: loaded /vendor/lib/egl/libGLESv2_emulation.so
W/.test.helloonn: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
W/.test.helloonn: Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
I/System.out: Get started Onnx!
I/System.out: set nnapi and opt level done!
W/onnxruntime:  [W:onnxruntime:ort-java, nnapi_execution_provider.cc:209 GetCapability] NnapiExecutionProvider::GetCapability, number of partitions supported by NNAPI: 10 number of nodes in the graph: 34 number of nodes supported by NNAPI: 11
I/TypeManager: Failed to read /vendor/etc/nnapi_extensions_app_allowlist ; No app allowlisted for vendor extensions use.
I/System.out: model ort done!
I/System.out: session run start
A/libc: Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 21821 (.test.helloonnx), pid 21821 (.test.helloonnx)
</pre>
I debugged the project and find the program crashed on OrtSession.class Line 145
`OnnxValue[] outputValues = this.run(OnnxRuntime.ortApiHandle, this.nativeHandle, this.allocator.handle, inputNamesArray, inputHandles, (long)inputNamesArray.length, outputNamesArray, (long)outputNamesArray.length, runOptionsHandle);`

I rewrite the model with single input node X:[1, 518], output node is Y:[1,466]. The program runs correctly. **I wonder that can onnxruntime-mobile support model inference with multiple inputs?** 

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows(X64), AVD: Android 11.0 x86, API 30, RAM:1536MB
- ONNX Runtime installed from (source or binary): onnxruntime-mobile-1.8.0.aar from maven
- ONNX Runtime version: onnxruntime-mobile-1.8.0
- Python version: 3.8
- CUDA/cuDNN version: CUDA: v11.1, cuDNN: 8.0.4
- GPU model: RTX2070 Super 

**To Reproduce**
- Export the model from pytorch by using torch.onnx.export() without dynamic axes setting
- Convert the onnx format to ort format by using python -m onnxruntime.tools.convert_onnx_models_to_ort, optimization_level is basic
-Import the onnxruntime-mobile-1.8.0.aar package,  loading the model in Android Studio project:
<pre>
OrtEnvironment env = OrtEnvironment.getEnvironment();  
OrtSession.SessionOptions opts = new SessionOptions();
try { 
     opts.addConfigEntry("session.load_model_format", "ORT");
     opts.addNnapi();
     opts.setOptimizationLevel(OptLevel.BASIC_OPT);
     System.out.println("set nnapi and opt level done!");
     OrtSession session =  env.createSession(model_path, opts);
     System.out.println("model ort done!");
     Map<String, OnnxTensor> mInputs = new HashMap<String, OnnxTensor>();
     mInputs.put("X1", OnnxTensor.createTensor(env, new float[1][466]));
     mInputs.put("X2", OnnxTensor.createTensor(env, new float[1][52]));
     System.out.println("session run start");
     Result result = session.run(mInputs);
     session.close();
     System.out.println("session run done!");
     float[][] out = (float[][]) result.get(0).getValue();
     } catch (OrtException e) {
            e.printStackTrace();
     }
</pre>
- Run the porject with AVD then got the program crash.

**Expected behavior**
Get the right output array with dimension as [1, 462]

**Comment by gwang-msft** [comment]
We need more information to investigate the crash. 
Please share the model, or let us know which model you were using if it is a pre-trained pytorch model.
Have you tried running the project without NNAPI, did it crash also?

**Comment by xyannly** [comment]
I tried to comment the "addNNapi()" and the program still crashed.
The model is given in [model](https://drive.google.com/file/d/1v686mSlweVwCAPJa8WJYrdbw11gjzwZ_/view?usp=sharing). 

I found another model with three input nodes in the onnxruntime release--[partial-inputs-test-2.onnx](https://github.com/microsoft/onnxruntime/blob/master/java/testdata/partial-inputs-test-2.onnx). I converted this model to ort format and tried to run the inference, the program still crashed.

The commad I used to do the conversion of ONNX format models to ORT format is:
`python -m onnxruntime.tools.convert_onnx_models_to_ort --optimization_level basic ./model_path`

**Comment by gwang-msft** [comment]
If you use `python -m onnxruntime.tools.convert_onnx_models_to_ort --optimization_level basic ./model_path` to convert the model to ort format, the model you shared will be converted to a 800MB model, and ORT will throw out of memory error, since the memory is a limited resource on mobile platform.

You may try to convert the model using `--optimization_level disable` to reduce the model size, unfortunately, the current pre-built onnxruntime-mobile does not support some operators in the model, you would need to custom build your own onnxruntime-mobile binary, please see https://www.onnxruntime.ai/docs/how-to/mobile/custom-build.html

**Comment by skottmckay** [comment]
Note that the model is growing due to some nodes being able to be constant folded (i.e. their output can be pre-calculated as all the inputs are known), with initializers being created for those output values. The worst offenders are these outputs which are well over 200MB each:

```
Name=59 Shape=[8, 32, 512, 466] NumElements=61079552 (output of Tile_28)
Name=129 Shape=[8, 32, 512, 512] NumElements=67108864 (output of Tile_64)
Name=199 Shape=[8, 32, 462, 512] NumElements=60555264 (output of Tile_100)
```

You may get lower memory usage by not allowing constant folding to take place by disabling the optimizations as suggested, as all those values may not need to be present at the same time when running the model. Cost of that is performance though as the values will get re-calculated every run instead of coming from initializers.

**Comment by xyannly** [comment]
Thanks for your replying, I checked the model for ort conversion, and I‘m sorry for giving the onnx format model with wrong input dimension. Here are the right mdoels with onnx format and basic.ort [model](https://drive.google.com/drive/folders/1mZHNq2wvTijQOWB00epB_ko5mk4KmaJ6?usp=sharing). Please try again. As for **partial-inputs-test-2.onnx**, I tried the model with code above and it still not work. If you can also run my model, there must be some errors in my code, could you please share the code?  Thanks a lot! 

Additionally, here are the configs in my andorid project:

**gradle.build**
![image](https://user-images.githubusercontent.com/40239008/122861443-10595100-d352-11eb-9ff0-0d1815d68192.png)

**AndroidManifest.xml**
![image](https://user-images.githubusercontent.com/40239008/122861688-6e863400-d352-11eb-93ba-c61272f9f8c1.png)

The MainAcitvity is 
![image](https://user-images.githubusercontent.com/40239008/122861870-c0c75500-d352-11eb-9662-1de782ec1cd8.png)

JDK version: 1.8

**Comment by gwang-msft** [comment]
There is a bug in our java API for 32bit Android system, a fix is in progress, to unblock, please try to use Android Emulator in x86_64 flavor.
Thanks for reporting this issue.

**Comment by gwang-msft** [comment]
The fix to this issue is merged, and the fix will be part of  onnxruntime-mobile-1.8.1 patch release.

**Comment by xyannly** [comment]
I tried to inference the model by using Android Emulator in x86_64 flavor, and it worked! Thanks a lot for your replying.
